src/parser/mage/fire/modules/features/CombustionCharges.tsx

```diff
@@ -65,6 +60,9 @@ class CombustionCharges extends Analyzer {
   }
 
   flagTimeline() {
+    if (!this.lastCastEvent) {
```

Comment:
`flagTimeline` is only called when this is true. In that case it would be cleaner to pass the lastCastEvent as an argument. Class fields make code harder to follow as they can be modified anywhere in a class, while parameters can make a method "pure" meaning it has no side-effects which makes it very easy to reason about. See the suggestions.